app/apphandlers/gatewayAppHandler.go

```diff
@@ -132,23 +132,26 @@ func (a *gatewayApp) StartRudderCore(ctx context.Context, options *app.Options)
 	})
 	drainConfigManager, err := drain_config.NewDrainConfigManager(config, a.log.Child("drain-config"))
 	if err != nil {
-		return fmt.Errorf("drain config manager setup: %v", err)
+		a.log.Errorw("drain config manager setup fail", "error", err)
 	}
-	defer drainConfigManager.Stop()
-	g.Go(misc.WithBugsnag(func() (err error) {
-		return drainConfigManager.DrainConfigRoutine(ctx)
-	}))
-	g.Go(misc.WithBugsnag(func() (err error) {
-		return drainConfigManager.CleanupRoutine(ctx)
+
+	drainConfigHttpHandler := http.Handler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "unable to start drain config http handler", http.StatusInternalServerError)
 	}))
```

Comment:
This can probably be moved to `drain_config` package.

```GO
func ErrorResponder() http.Handler {
	return http.Handler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		http.Error(w, "unable to start drain config http handler", http.StatusInternalServerError)
	}))
}
```

Consumer can use this as:
```GO
drainConfigHttpHandler := drain_config.ErrorResponder()
```